easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,92 @@
+##
+# Copyright 2020 NVIDIA
+#
+# This file is triple-licensed under GPLv2 (see below), MIT, and
+# BSD three-clause licenses.
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'syshost': ['hardcode:cluster', "System name", CUSTOM],
+            'transmission': ['syslog', "", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # Remove the subdir from the install prefix to resolve the
+        # following configure error.  XALT automatically creates a
+        # versioned directory hierarchy.
+        #
+        #    Do not install XALT with 2.7.27 as part of the
+        #    prefix.  It will lead to many many problems with future
+        #    installs of XALT
+        #
+        #    Executables built with the current version of XALT
+        #    will not work with future installs of XALT!!!
+        #
+        #    If you feel that you know better than the developer
+        #    of XALT then you can configure XALT with the configure
+        #    --with-IrefuseToInstallXALTCorrectly=yes and set the prefix
+        #    to include the version
+        self.installdir = os.path.normpath(re.sub(self.install_subdir, '', self.installdir))
```

Comment:
After discussing this with @rtmclay, I think it makes sense to wait until there's an XALT release that has support for picking up on `$XALT_DIR` (see @samcmill's PR to add that at https://github.com/xalt/xalt/pull/26).

If XALT picks up on `$XALT_DIR`, none of this trickery with the installation prefix is needed anymore, and we can have several XALT installations side by side. As long as `$XALT_DIR` points to an existing XALT installation when static binaries that have been hijacked by XALT are used, XALT tracking will still work.

I suggested to @rtmclay to let XALT log a message to syslog when it can't find its tool anymore (which is what `$XALT_DIR` would be for, and which is why XALT is currently so pedantic about where it's being installed), which is a decent way of signalling that XALT tracking got broken because XALT's tool were (re)moved.

Reply:
Makes sense to me.

When XALT no longer requires special installation prefix handling, then I think a dedicated XALT EasyBlock will not be necessary?

Reply:
It would still be useful I think, w.r.t. the required values for `config_py` & co...

Reply:
Addressed the original comment, so making this as resolved.